lib/cipherstash/index/dynamic_filter_match.rb

```diff
@@ -0,0 +1,50 @@
+require_relative "./bloom_filter"
+
+module CipherStash
+    class Index
+      # Implementation of the dynamic-filter-match index
+      #
+      # @private
+      class DynamicFilterMatch < Index
+        INDEX_OPS = {
+          "match" => -> (idx, s) do
+            filter = BloomFilter.new(
+              [idx.meta_settings["$filterKey"]].pack("H*"),
+              idx.mapping_settings
+            )
+
+            bits = idx.text_processor.perform(s)
+              .reduce(filter) { |filter, term| filter.add(term) }
```

Comment:
Stylistic thing. While Rusaceans do it like this, Rubyists tend to do:

```ruby
.reduce { |filter, term|
  filter.add(term)
}.bits.to_a
```

But now I'm picking nits :p

Reply:
What bong-ripping Rubyists do _you_ hang out with?  I'd expect to see it written it the way that Drew has done.  If you're going to multi-line it for some reason, it'd be `do`/`end` rather than multi-line braces.

Reply:
Oh lordy. Not these debates again haha! I'm not going to die on this hill :P

Reply:
I ended up removing this and using `.each` instead: https://github.com/cipherstash/ruby-client/pull/13/commits/a2a8adf6073082c942f3e1d6f63b24a04850faac#diff-b76c2fd0d2a1d587276d59cbe48470e48cadb5b9aec9f0a762d08f1b064c83f3R69